### PR TITLE
Added error handling using Match Statements

### LINKED ISSUE #8: Add a progress bar using Indicatif  module

### LINKED ISSUE #9: Introduce better error handling mechanism

### PR DESCRIPTION
**Pull request by iamakshat01**
1. Closes #9 , done using match statements

### CODE REVIEW

File: src/ops/core_ops.rs [hunk 1]
```diff
@@ -24,7 +31,26 @@ pub fn handle_core_command(core_command: CoreCommand, program: &Program) {
     let command = core_command.command;
     match command {
         CoreSubcommand::Fetch(fetch_tranche) => {
-            let account: TrancheConfig = program.account(fetch_tranche.tranche_id).expect("Could not find tranche with given publickey");
+            let bar = ProgressBar::new(1);
+            bar.set_style(ProgressStyle::with_template("[{elapsed_precise}] {bar:40.cyan/blue} {pos:>7}/{len:7} {msg}")
+            .unwrap()
+            .progress_chars("##-"));
+            let account:Result<TrancheConfig,ClientError> = program.account(fetch_tranche.tranche_id);
+            bar.finish();
```
Comment (giacomo-cc) [inline src/ops/core_ops.rs]: I don't think this is how the progress bar should be used. probably we should run the account fetch on a dedicated thread and in parallel update the bar value. the real problem is that as of now we 're missing a concept of progress in the account fetch we right now move from:
`not fetched -> fetched`
with no concept of
`not fetched -> fetched at 10% -> fetched 20% -> ... -> fetched`

So we could simulate a fetching progress or make the progress bar indeterminate (endless loading until the account is fetched). Both option are valid, but considering the state of the project I'll keep this as a future issue and we can remove the progress bar from this PR.

**Review by giacomo-cc — CHANGES_REQUESTED** [review]
the error handling is perfect, we should only remove the progress bar as stated in the comment

File: src/ops/core_ops.rs (hunk 1 above)
Comment (iamakshat01) [reply src/ops/core_ops.rs]: Ok, will read about multithreadng in rust and come up with it in next PR.

**Review by giacomo-cc — APPROVED** [review]

File: src/ops/core_ops.rs (hunk 1 above)
Comment (giacomo-cc) [reply src/ops/core_ops.rs]: @iamakshat01 for now we can keep on hold multi threading considering that has lower priority. let's discuss this in later catchup

**Pull request merged**